src/app/categories/state/entry-categories.service.ts

```diff
@@ -42,14 +42,13 @@ export class EntryCategoriesService {
   /**
    * Updates the list of categories for the specified entry.
    */
-  updateEntryCategories(entryId: number) {
+  updateEntryCategories(entryId: number, published: boolean) {
     this.entryCategoriesStore.setLoading(true);
     this.entryCategoriesStore.setError(false);
     this.entryCategoriesStore.remove();
     this.currentSubscription?.unsubscribe();
```

Comment:
In this code, `currentSubscription` isn't always set to a new value each time through this function, it may remain set to the old value if `published` is false.  So, it'd probably be good to either:
* confirm via testing that `unsubscribe` can be called multiple times on a given subscription, or
* set `currentSubscription` to `null` after the `unsubscribe`.

Reply:
I opted to set the currentSubscription to null after the unsubscribe.